src/main/java/seedu/address/model/person/Contact.java

```diff
@@ -0,0 +1,108 @@
+package seedu.address.model.person;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.address.model.tag.Tag;
+import seedu.address.model.tag.UniqueTagList;
+
+/**
+ * Represents a Person in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Contact extends Person {
```

Comment:
Since `Person` is still a class and not an interface, probably do not need to repeat all methods or variables due to inheritance. 
In addition, is this class really needed? Seems to me that a simple new variable of `Type` class would suffice in the original `Person` class.